.github/workflows/main.yml

```diff
@@ -57,37 +58,57 @@ jobs:
         else
           echo "::set-output name=result::odd"
         fi
+  build-and-cache-node-modules:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Setup repo
+        uses: actions/checkout@v3
+         
+      - uses: actions/setup-node@v3
+        with: 
+          registry-url: ${{ env.NPM_REGISTRY }}
+          scope: "${{ env.SCOPE }}"
+          cache: 'npm'
+
+      - name: Setup Deno
+        uses: denoland/setup-deno@v1
+        with:
+          deno-version: "1.23.4"
+
+      - name: Build all
+        run: |
+          export PATH="$PATH:$(pwd)"
+          npm run build-all
+
+      - name: Cache node modules
+        uses: actions/cache@v3
+        with:
+          path: ./node_modules/**
+          key: cache-node-modules-${{ github.run_id }}
```

Comment:
I was not sure of this. Does the `package-lock.json` file change if I make a change to a file like `taqueria-plugin-ligo/compile.ts` for example?

Reply:
It does not. The package-lock.json file would get updated whenever you modify a package.json file and run `npm update` (which is what we all should be doing if we modify the package.json file for any package).

I suppose why you ask that, is because the node_modules folder also contains packages we've developed, not just third-party modules. However, these should be installed via symlink and so the symlink should still point to the same place.

When we cache node_modules, are the file types of these symlinks preserved as a link, or are they resolved first and then cached?

Reply:
I think they are resolved and then cached as I am only caching the `node_modules` in the project root and all compiled files in the plugin directories are available in the cache